### PR TITLE
Standardized Plugins Downloads and Center Icons

### PR DESCRIPTION
**Pull request by yiliang114**
fix(vsx-registry): Standardized Plugins Downloads and Center Icons
Signed-off-by: yiliang114 <1204183885@qq.com>

#### What it does

1. Standardized plugins downloads.
2. Downloads icon and reviews icon are not center a little, and it center icons of `VSXExtensionsWidget`. 

#### How to test

1. Start application and turn `Extensions` activity bar.
2. Search something such as `chinese`.

image

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers
In VS Code, the downloads of extension is formatted with `en-US` and it endWiths `k` no matter what language is set.  And the reviews number is always kept as an `Int`.

image

https://github.com/eclipse-theia/theia/blob/0c0f8c7a7aedd67930bf65d6f63d827721707d60/packages/vsx-registry/src/browser/vsx-extension.tsx#L446

But if theia uses `undefined` to new `Intl.NumberFormat`, it will cause language inconsistency when my browser default language is different from theia's default language.

image

After modification:
image

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by yiliang114** [comment]
Why sometimes CI runs failed after long time? And if i try to rerun them which have ended with some errors, they are okey again.

**Comment by msujew** [comment]
@yiliang114 We have some flaky tests in there which sometimes (randomly) like to fail, usually related to file access which just times out. See https://github.com/eclipse-theia/theia/issues/10348.

**Review by msujew — APPROVED** [review]
Thanks, the changes look good to me 👍

**Review by vince-fugnitto — CHANGES_REQUESTED** [review]

File: packages/vsx-registry/src/browser/vsx-extension.tsx [hunk 1]
```diff
@@ -466,7 +466,7 @@ export class VSXExtensionComponent<Props extends VSXExtensionComponent.Props = V
                     </div>
                     <div className='stat'>
                         {!!downloadCount && <span className='download-count'><i className={codicon('cloud-download')} />{downloadCompactFormatter.format(downloadCount)}</span>}
-                        {!!averageRating && <span className='average-rating'><i className={codicon('star-full')} />{averageRating.toFixed(1)}</span>}
+                        {!!averageRating && <span className='average-rating'><i className={codicon('star-full')} />{averageRating.toFixed(0)}</span>}
```
[the remark this answers is not in the document]
Comment (yiliang114) [reply packages/vsx-registry/src/browser/vsx-extension.tsx:469]: It seems that I think it is too simple... Let me take a look at vscode's count processing logic, i will solve it right now.

[the remark this answers is not in the document]
Comment (yiliang114) [reply packages/vsx-registry/src/browser/vsx-extension.tsx:469]: It seems that I think it is too simple... Let me take a look at vscode's count processing logic, i will solve it right now.

[the remark this answers is not in the document]
Comment (yiliang114) [reply packages/vsx-registry/src/browser/vsx-extension.tsx:469]: It seems that I think it is too simple... Let me take a look at vscode's count processing logic, i will solve it right now.

[the remark this answers is not in the document]
Comment (yiliang114) [reply packages/vsx-registry/src/browser/vsx-extension.tsx:469]: I saw here in fact, but I mistakenly thought that the correct number should be displayed at that time. And I will solve it again in a minute to keep the data on both sides consistent.

File: packages/vsx-registry/src/browser/vsx-extension.tsx [hunk 2]
```diff
@@ -527,7 +537,7 @@ export class VSXExtensionEditorComponent extends AbstractVSXExtensionComponent {
             </div>
             {
                 sanitizedReadme &&
-                < div className='scroll-container'
+                <div className='scroll-container'
```
Comment (yiliang114) [inline packages/vsx-registry/src/browser/vsx-extension.tsx:540]: Is this an error caused by the previous formatting?  I think there should be no spaces here.

**Comment by yiliang114** [comment]
What it looks like now is 
image

File: packages/vsx-registry/src/browser/vsx-extension.tsx [hunk 3]
```diff
@@ -516,7 +523,10 @@ export class VSXExtensionEditorComponent extends AbstractVSXExtensionComponent {
                         </span>
                         {!!downloadCount && <span className='download-count' onClick={this.openExtension}>
                             <i className={codicon('cloud-download')} />{downloadFormatter.format(downloadCount)}</span>}
-                        {averageRating !== undefined && <span className='average-rating' onClick={this.openAverageRating}>{this.renderStars()}</span>}
+                        {
+                            averageRating !== undefined &&
+                            <span className='average-rating' title={getAverageRatingTitle(averageRating)} onClick={this.openAverageRating}>{this.renderStars()}</span>
```
[the remark this answers is not in the document]
Comment (yiliang114) [reply packages/vsx-registry/src/browser/vsx-extension.tsx:528]: What it looks like now is 
image

File: packages/vsx-registry/src/browser/vsx-extension.tsx [hunk 4]
```diff
@@ -443,7 +443,14 @@ export namespace AbstractVSXExtensionComponent {
 
 const downloadFormatter = new Intl.NumberFormat();
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-const downloadCompactFormatter = new Intl.NumberFormat(undefined, { notation: 'compact', compactDisplay: 'short' } as any);
+const downloadCompactFormatter = new Intl.NumberFormat('en-US', { notation: 'compact', compactDisplay: 'short' } as any);
+const averageRatingFormatter = (averageRating: number): number => Math.round(averageRating * 2) / 2;
+const getAverageRatingTitle = (averageRating: number | undefined): string => {
```
Comment (vince-fugnitto) [inline packages/vsx-registry/src/browser/vsx-extension.tsx:448]: Is there a case where `getAverageRatingTitle` is called where `averageRating` is `undefined`?

File: packages/vsx-registry/src/browser/vsx-extension.tsx [hunk 5]
```diff
@@ -443,7 +443,14 @@ export namespace AbstractVSXExtensionComponent {
...
+    if (!averageRating) {
+        return '';
+    }
+    return `Average rating: ${averageRatingFormatter(averageRating)} out of 5`;
```
Comment (vince-fugnitto) [inline packages/vsx-registry/src/browser/vsx-extension.tsx:452]: The following should actually be localized. The same is true for other items in the `get tooltip()` if you're up for it :)

```suggestion
    return nls.localize('theia/vsx-extension/averageRating', 'Average rating: {0} out of 5', averageRatingFormatter(averageRating);
```

File: packages/vsx-registry/src/browser/vsx-extension.tsx (hunk 4 above)
Comment (yiliang114) [reply packages/vsx-registry/src/browser/vsx-extension.tsx]: This case that I considered before will not exist and has been removed.

**Comment by yiliang114** [comment]
image


Referring to the format of vscode's Chinese translation, I added a Chinese translation to theia. Now its performance in Chinese and English is respectively

https://github.com/eclipse-theia/theia/blob/f46789c47e0dd8ba0069c66084a5808716ad07c9/packages/core/i18n/nls.zh-cn.json#L394

image
image

File: packages/vsx-registry/src/browser/vsx-extension.tsx (hunk 5 above)
Comment (vince-fugnitto) [reply packages/vsx-registry/src/browser/vsx-extension.tsx]: @yiliang114 I also mentioned `get tooltip()`, do you mind updating the code there to add translations? The tooltip is used when hovering over extensions in the view itself:

Screen Shot 2022-08-03 at 2 17 26 PM

https://github.com/eclipse-theia/theia/blob/5aeef6c0c683b4e91713ab736957e6655b486adc/packages/vsx-registry/src/browser/vsx-extension.tsx#L273-L289

File: packages/core/i18n/nls.json [hunk 6]
```diff
@@ -390,6 +390,9 @@
         "update-version-version-error": "Failed to install version {0} of {1}."
       }
     },
+    "vsx-extension": {
+      "averageRating": "Average rating: {0} out of 5"
+    },
```
Comment (msujew) [inline packages/core/i18n/nls.json:395]: Can you please reset this change and rebase onto latest `master`? That should then automatically create the appropriate translation entries. You can then change the `zh-cn` translation if you don't find it fitting.

File: packages/vsx-registry/src/browser/vsx-extension.tsx [hunk 7]
```diff
@@ -443,7 +443,10 @@ export namespace AbstractVSXExtensionComponent {
 
 const downloadFormatter = new Intl.NumberFormat();
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-const downloadCompactFormatter = new Intl.NumberFormat(undefined, { notation: 'compact', compactDisplay: 'short' } as any);
+const downloadCompactFormatter = new Intl.NumberFormat('en-US', { notation: 'compact', compactDisplay: 'short' } as any);
+const averageRatingFormatter = (averageRating: number): number => Math.round(averageRating * 2) / 2;
+const getAverageRatingTitle = (averageRating: number): string =>
+    nls.localize('theia/vsx-extension/averageRating', 'Average rating: {0} out of 5', averageRatingFormatter(averageRating));
```
Comment (msujew) [inline packages/vsx-registry/src/browser/vsx-extension.tsx:449]: Minor: This should likely be
```suggestion
    nls.localize('theia/vsx-registry/averageRating', 'Average rating: {0} out of 5', averageRatingFormatter(averageRating));
```

File: packages/core/i18n/nls.json (hunk 6 above)
Comment (yiliang114) [reply packages/core/i18n/nls.json:395]: @msujew Thank you, but I have a confusion. Do you mean that I can divide it into two steps to do localization?

In the first step, I only need to use `nls.localize('theia/vsx-extension/averageRating', 'Average rating: {0} out of 5', averageRatingFormatter(averageRating)` in the code. CI will automatically generate the corresponding entries in' nls.json' and' nls-zh-cn.json '. 

In the second step, if the default Chinese translation is not very suitable, then I modify the Chinese translation in' nls-zh-cn.json?

File: packages/vsx-registry/src/browser/vsx-extension.tsx (hunk 5 above)
Comment (yiliang114) [reply packages/vsx-registry/src/browser/vsx-extension.tsx]: @vince-fugnitto I just took care of `get tooltip()`. I have used `nls.localize` to repalce string of `Publisher`, `License` and `Download count` here. 

By the way, as @msujew said below, I have reset the code of `nls.json` and `nls.zh-cn.json`,  Waiting for the code to be incorporated into master and waiting CI will automatically generate the appropriate translation entries. Is it the right thing to do for me?

**Review by vince-fugnitto — APPROVED** [review]
The changes look good to me 👍  The new localizations will be added to the framework before the release as they do not work from a fork and we changed our approach.

**Comment by yiliang114** [comment]
> The changes look good to me 👍 The new localizations will be added to the framework before the release as they do not work from a fork and we changed our approach.

Thank you both @msujew @vince-fugnitto

**Pull request merged**